src/azure-account.ts

```diff
@@ -377,8 +381,21 @@ export class AzureLoginHelper {
 			const config = workspace.getConfiguration('azure');
 			if (config.get('cloud') !== selected.environment.name) {
 				this.doLogin = true;
```

Comment:
Move this.doLogin to the top as before, think this should be good to resolve Rachel's comment.